server/src/main/java/org/elasticsearch/index/shard/SimpleShardStats.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.shard;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.core.Nullable;
+
+import java.io.IOException;
+
+public class SimpleShardStats implements Writeable, ToXContentFragment {
```

Comment:
I think we should name this `ShardCountStats` to avoid the word "Simple" in the name. Often things that start simple may end up becoming non-simple in the future, and it's kind of like naming something "FastMyClass" and then having to go with "ReallyFastMyClass" down the road.